cloud-controller-manager/do/metadata.go

```diff
@@ -20,12 +20,17 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"os"
 )
 
 const dropletRegionMetadataURL = "http://169.254.169.254/metadata/v1/region"
 
 // dropletRegion returns the region of the currently running program.
 func dropletRegion() (string, error) {
+	// FAKE_REGION can be specified for local testing.
+	if region := os.Getenv("FAKE_REGION"); region != "" {
+		return region, nil
+	}
```

Comment:
Does it mean that if someone sets environment variable in production environment, presumably by mistake, a region will not be detected correctly? Would it be better to pass an environment identifier in a command line, like `--test-env`, so that the code execution paths do not rely on presence or absence of arbitrary envs? :thinking:

Reply:
Adding a custom flag has proven to be far from trivial in CCM since it's tightly bundled to the (fairly complex) flag scaffolding that every CCM inherits from kube-controller-manager. I haven't had the bandwidth yet to look more deeply into it, but I'd be very open to switching to flags if you could make it happen somehow.

I can make a quick amendment to the PR to log when the environment flag is used as an additional warning. I also tried to document its usage, so hopefully accidental presence of the environment variable in production environments should be minimized.

Reply:
Here's another idea I've had: instead of specifying the region, it'd be cool to have a mock HTTP server that returns a fake region. The challenge would be to make it respond on the metadata IP address though.

Just wanted to mention it in case you or anyone else feels adventurous to prototype the idea.

Reply:
Thanks for adding the log!